core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -175,6 +182,51 @@ class SYCLTeamMember {
     m_item.barrier(sycl::access::fence_space::local_space);
   }
 
+  // FIXME_SYCL move somewhere else and combine with other places that do
+  // parallel_scan Exclusive scan returning the total sum.
+  template <typename Type>
+  static Type prescan(sycl::nd_item<2> m_item, Type* temp, int n) {
```

Comment:
Without the reference to Mark Harris' writeup (I assume that's where you took that code from) this is hardly readable.

Reply:
This really is the usual parallel scan algorithm that we also use in the other places. I'll add a reference.